xarray/core/concat.py

```diff
@@ -459,6 +459,9 @@ def ensure_common_dims(vars):
             combined = concat_vars(vars, dim, positions)
             assert isinstance(combined, Variable)
             result_vars[k] = combined
+        else:
+            # preserves original variable order
+            result_vars[k] = result_vars.pop(k)
```

Comment:
I'm wondering: is it possible for `k` to not already be in `result_vars`? If it is, this might add `None` valued entries to `result_vars`, so we'd have to check for `None` before assigning.

Reply:
That's what I was wondering too. Might not use `else` but `elif`?

```python
elif k in result_vars:
    result_vars[k] = result_vars.pop(k)
```

Reply:
actually, it seems those non-dimension coordinates that would not be in `concat_over` are in `variables_to_merge` and in `dim_coords`, so I guess accidentally calling `pop` for a variable not in `result_vars` is not possible?

Reply:
@keewis I've changed from `else` to `elif` as written above. This only re-inserts into `result_vars` if already in. Thanks for catching this!

Reply:
well, as I said I'm totally not sure anymore that we really have to check, but I guess the check itself is pretty fast. Let's get the opinion of someone else?

Reply:
> so I guess accidentally calling `pop` for a variable not in `result_vars` is not possible?

Sorry, your message was first displayed after mine was sent. So `else` would have worked without problems? Maybe the `elif` is more clear, that it should already be availble in `result_vars`.

Reply:
> Let's get the opinion of someone else?

Would be very much appreciated. Also not sure, if additional tests are needed here?

Reply:
> Maybe the `elif` is more clear, that it should already be available in `result_vars`.

it is, but a comment would also work, I guess.

> Also not sure, if additional tests are needed here?

I think a test that makes sure the coordinates stay in order (data variables seem to always be in `concat_over`) would be good. That way we can be confident about not messing this up again.

Reply:
I don't think that check is required but seems fine to be safe.

Reply:
OK, then I'll leave it in for now...